packages/eyes/src/eyes-test/eyes.ts

```diff
@@ -0,0 +1,168 @@
+import * as path from 'path';
+import * as fs from 'fs';
+import glob from 'glob';
+import chalk from 'chalk';
+import {JSDOM} from 'jsdom';
+import {consoleLog} from '@ui-autotools/utils';
+import {parseFilename} from '../generate-snapshots/filename-utils';
+
+interface IResult {
+  name: string;
+  status: 'error' | 'new' | 'modified' | 'unmodified';
+  url: string;
+  error?: any;
+}
+
+function getGridClientConfig(projectPath: string) {
+  const projectName = require(path.join(projectPath, 'package.json')).name;
+  if (!projectName) {
+    throw new Error('The project should have a package.json file with a "name" field.');
+  }
+
+  if (!process.env.APPLITOOLS_API_KEY) {
+    throw new Error('The environment variable "APPLITOOLS_API_KEY" needs to be defined.');
+  }
+
+  const branchName = projectName + '/master';
+  const viewportWidth = 800;
+  const viewportHeight = 600;
+
+  return {
+    appName: projectName,
+    branchName,
+    batchName: projectName,
+    browser: {
+      name: 'chrome',
+      width: viewportWidth,
+      height: viewportHeight
+    }
+  };
+}
+
+interface IResource {
+  url: string;
+  type: string;
+  value: Buffer;
+}
+
+function getStaticResources(cssFilenames: string[], resourceDir: string): {[url: string]: IResource} {
+  const resources: {[url: string]: IResource} = {};
+  for (const cssFilename of cssFilenames) {
+    const url: string = '/' + cssFilename;
+    resources[url] = {
+      url,
+      type: 'text/css',
+      value: fs.readFileSync(path.join(resourceDir, cssFilename))
+    };
+  }
+  return resources;
+}
+
+function logEyesResult({name, status, url, error}: IResult) {
+  const {compName, simName, styleName} = parseFilename(name, '.snapshot.html');
+
+  const isModified = status === 'modified';
+  const isNew = status === 'new';
+  const isError = status === 'error';
+
+  const formattedUrl = isModified ? `${chalk.cyan('URL')}: ${chalk.underline(url)}` : '';
+  const statusMessage = isModified ? chalk.red('MODIFIED') :
+               isNew ? chalk.green('NEW') :
+               isError ? chalk.bgRedBright('ERROR') :
+               chalk.green('UNMODIFIED');
+  consoleLog(`${statusMessage} ${chalk.bold(compName)}: ${simName}. Style: ${styleName}. ${formattedUrl}`);
+
+  if (isError) {
+    consoleLog(error);
+  }
+}
+
+// We cannot allow any of the result promises to reject, because we're going to
+// await on a bunch of them in parallel using Promise.all, and a single rejected
+// promise would reject the entire batch.
+function getTestResult(testName: string, testResult: Promise<any>): Promise<IResult> {
+  return (
+    testResult
+    .catch((err: any) => err)
+    .then((res: any) => res instanceof TestFailedError ? res.getTestResults() : res)
+    .then((res: any) => Array.isArray(res) && res[0] instanceof TestResults ? res[0] : res)
+    .then((res: any): IResult => {
+      const url = res.getUrl();
```

Comment:
This method is only available on instances of `TestResults` and should only be used in the `if` below.